### PR TITLE
Nm notification preferences

### PR DESCRIPTION
**Pull request by nickmcmahon01**
Create the boilerplate and the GET endpoint of the snooze preference.

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/repository/SnoozePreferenceRepository.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.repository
+
+import org.springframework.data.repository.CrudRepository
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.model.SnoozePreference
+import java.time.LocalDate
+import java.util.*
+
+@Repository
+interface SnoozePreferenceRepository : CrudRepository<SnoozePreference, UUID> {
```
Comment (jonhindle) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/repository/SnoozePreferenceRepository.kt:10]: Is the use of `SnoozePreference` and `SnoozePreferences` in various naming bits an intentional differentiation?

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/model/SnoozePreferenceTest.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.model.SnoozePreference
+import java.time.LocalDate
+
+class SnoozePreferenceTest {
+
+    @Test
+    fun `Should return a valid date`() {
+        val quantumId = "XYZ"
+        val date = LocalDate.now()
+        val snoozePref = SnoozePreference(quantumId, date)
```
Comment (jonhindle) [inline src/test/java/uk/gov/justice/digital/hmpps/cmd/api/model/SnoozePreferenceTest.kt:14]: I think you can de-structure these 😬 
```
val (quantumId, snoozeUntil)  = SnoozePreference(quantumId, date)
```

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/SnoozePreferencesController.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.dto.PreferenceDto
+import uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.dto.UpdateSnoozeRequest
+import uk.gov.justice.digital.hmpps.cmd.api.uk.gov.justice.digital.hmpps.cmd.api.service.SnoozeNotificationPreferenceService
+
+@Api(tags = ["notification-preferences"])
+@RestController
+@RequestMapping(produces = [APPLICATION_JSON_VALUE])
+class SnoozePreferencesController(val snoozeNotificationPreferenceService: SnoozeNotificationPreferenceService) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/SnoozePreferencesController.kt]: I have renamed.

File: src/test/java/uk/gov/justice/digital/hmpps/cmd/api/model/SnoozePreferenceTest.kt (hunk 2 above)
Comment (nickmcmahon01) [reply src/test/java/uk/gov/justice/digital/hmpps/cmd/api/model/SnoozePreferenceTest.kt]: I'd prefer to be verbose on unit tests.

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/repository/SnoozePreferenceRepository.kt (hunk 1 above)
Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/hmpps/cmd/api/repository/SnoozePreferenceRepository.kt]: I have renamed to userPreference consistently everywhere

**Pull request merged**